src/gfn/samplers.py

```diff
@@ -89,38 +93,37 @@ def sample_actions(
     def sample_trajectories(
         self,
         env: Env,
-        off_policy: bool,
         states: Optional[States] = None,
         n_trajectories: Optional[int] = None,
-        debug_mode: bool = False,
+        save_estimator_outputs: bool = False,
+        save_logprobs: bool = True,
         **policy_kwargs,
     ) -> Trajectories:
         """Sample trajectories sequentially.
 
         Args:
             env: The environment to sample trajectories from.
-            off_policy: If True, samples actions such that we skip log probability
-                calculation, and we save the estimator outputs for later use.
             states: If given, trajectories would start from such states. Otherwise,
                 trajectories are sampled from $s_o$ and n_trajectories must be provided.
             n_trajectories: If given, a batch of n_trajectories will be sampled all
                 starting from the environment's s_0.
+            save_estimator_outputs: If True, the estimator outputs will be returned. This
+                is useful for off-policy training with tempered policy.
+            save_logprobs: If True, calculates and saves the log probabilities of sampled
+                actions. This is useful for on-policy training.
             policy_kwargs: keyword arguments to be passed to the
                 `to_probability_distribution` method of the estimator. For example, for
                 DiscretePolicyEstimators, the kwargs can contain the `temperature`
                 parameter, `epsilon`, and `sf_bias`. In the continuous case these
                 kwargs will be user defined. This can be used to, for example, sample
                 off-policy.
-            debug_mode: if True, everything gets calculated.
```

Comment:
Why is `debug_mode` removed? If I recall, this was important for tests.

Reply:
Isn't this the same as `recalculate_all`?

Reply:
right -- I'll change it back and add a note :)